components/edit-collective/Form.js

```diff
@@ -473,6 +483,65 @@ class EditCollectiveForm extends React.Component {
     }
   }
 
+  getApplicableTaxesFields = () => {
+    const { intl } = this.props;
+    const { collective } = this.state;
+    const fields = [];
+    const country = get(collective, 'location.country') || get(collective.host, 'location.country');
+    const taxes = getApplicableTaxesForCountry(country);
+
+    if (taxes.includes(TaxType.VAT)) {
+      fields.concat([
```

Comment:
Use push and drop the brackets, concat won't mutate `fields`.
```suggestion
      fields.push({...}, {...})
```

Reply:
Well spotted! This was indeed introducing a bug